bin/tx-prover/README.md

```diff
@@ -1,83 +1,83 @@
 
 ```sh
-cargo install miden-tx-prover-worker --locked
-cargo install miden-tx-prover-proxy --locked
+cargo install miden-tx-prover-cli --locked
 ```
 
 This will install the latest official version of the prover. You can install a specific version using `--version <x.y.z>`:
 
 ```sh
 cargo install miden-tx-prover-worker --locked --version x.y.z
 cargo install miden-tx-prover-proxy --locked --version x.y.z
-```
-
-Both services can be installed at once from the source code using specific git revisions with `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only.
+``` -->
 
-If you want to build from a local version, from the root of the workspace you can run:
+To build the CLI from a local version, from the root of the workspace you can run:
 
 ```bash
-make install-prover-worker
-make install-prover-proxy
+make install-prover-cli
 ```
 
+The CLI can be installed from the source code using specific git revisions with `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only.
+
 Note that for the prover worker you might need to enable the `testing` feature in case the transactions were executed with reduced proof-of-work requirements (or otherwise, the proving process will fail). This step will also generate the necessary protobuf-related files. You can achieve that by generating the binary with:
 
 ```bash
-make install-prover-worker-testing
+make install-prover-cli-testing
 ```
 
-## Running the Service
+## Usage
 
-Once installed, you can run the worker with:
+Once installed, you need to initialize the CLI with:
 
 ```bash
-RUST_LOG=info miden-tx-prover-worker
+miden-tx-prover-cli init
 ```
 
-By default, the server will start on `0.0.0.0:50051`. You can change this and the log level by setting the following environment variables:
+This will create the `miden-prover-service.toml` file in your current directory. This file will hold the configuration for both workers and the proxy. You can modify the configuration changing the host and ports of the services, and add workers. An example of a valid configuration is:
 
-```bash
-PROVER_SERVICE_HOST=<your-host>
-PROVER_SERVICE_PORT=<your-port>
-RUST_LOG=<log-level>
-```
+```toml
+[[workers]]
+host = "0.0.0.0"
+port = 8080
 
-And to run the proxy:
+[[workers]]
+host = "0.0.0.0"
+port = 8081
 
-```bash
-RUST_LOG=info miden-tx-prover-proxy
+[proxy]
+host = "0.0.0.0"
+port = 8082
 ```
 
-By default, the server will start on `0.0.0.0:6188`. This can be changed by setting:
+To add more workers, you will need to add more items with the `[[workers]]` tags.
+
+To start the worker service, once you have configurated at least one in the config file, you will need to run:
 
 ```bash
-PROXY_HOST=<your-host>
-PROXY_PORT=<your-port>
+RUST_LOG=info miden-tx-prover-cli start-worker
```

Comment:
Also, on the worker logs, maybe we should show which worker is the one printing the log (i.e. which port). If we change it so that only one worker is running then it wouldn't make much of a difference.

Reply:
This will no longer happen since now we are only running one worker.